### PR TITLE
Optimize learning returns curve

### PR DESCRIPTION
**Pull request by maksim-tolo**

### CODE REVIEW

File: contracts/BattleController.sol [hunk 1]
```diff
@@ -199,13 +200,18 @@ contract BattleController is Upgradable {
...
-        uint8 _factor;
-
-        if (_dragonsAmount < 15000) _factor = 20;
-        else if (_dragonsAmount < 30000) _factor = 10;
-        else _factor = 5;
-
-        return _hatchingPrice.mul(_factor).div(PERCENT_MULTIPLIER);
+        uint32 _factor;
+
+        if (_dragonsAmount < 3000) _factor = 2000000; // Hydrogen Eon
```
Comment (maxaleks) [inline contracts/BattleController.sol]: `MAX_GOLD_REWARD_MULTIPLIER` is less than the `_factor` value. In this case we will pay ```1000 * 2000000 / 100000 =  20000``` gold to the winner

Comment (maxaleks) [inline contracts/BattleController.sol]: `MAX_GOLD_REWARD_MULTIPLIER` is less than the _factor value. In this case we will pay `1000 * 2000000 / 100000 = 20000` gold to the winner. Or am I missing something?

Comment (xhipster) [reply contracts/BattleController.sol]: Hm, @maksim-tolo Could you check it? Its unexpected

Comment (maksim-tolo) [reply contracts/BattleController.sol]: Fixed

**Pull request merged**